pkg/segment/results/mresults/tests/metricsresults_test.go

```diff
@@ -404,3 +406,167 @@ func TestCalculateInterval(t *testing.T) {
 		}
 	}
 }
+
+func Test_GetResults_GTR(t *testing.T) {
+	test_GetResults_Ops(t,
+		map[uint32]float64{
+			0:     45,
+			10800: 1045,
+		},
+		map[uint32]float64{},
+		[]structs.QueryArithmetic{
+			{
+				LHS:        1,
+				ConstantOp: true,
+				Operation:  utils.LetGreaterThan,
+				Constant:   1045,
+			},
+		})
+}
+
+func Test_GetResults_GTE(t *testing.T) {
+	test_GetResults_Ops(t,
+		map[uint32]float64{
+			0:     45,
+			10800: 1045,
+		},
+		map[uint32]float64{
+			10800: 1045,
+		},
+		[]structs.QueryArithmetic{
+			{
+				LHS:        1,
+				ConstantOp: true,
+				Operation:  utils.LetGreaterThanOrEqualTo,
+				Constant:   100,
+			},
+		})
+}
+
+func Test_GetResults_LSS(t *testing.T) {
+	test_GetResults_Ops(t,
+		map[uint32]float64{
+			0:     45,
+			10800: 1045,
+		},
+		map[uint32]float64{
+			0: 45,
+		},
+		[]structs.QueryArithmetic{
+			{
+				LHS:        1,
+				ConstantOp: true,
+				Operation:  utils.LetLessThan,
+				Constant:   1045,
+			},
+		})
+}
+
+func Test_GetResults_LTE(t *testing.T) {
+	test_GetResults_Ops(t,
+		map[uint32]float64{
+			0:     45,
+			10800: 1045,
+		},
+		map[uint32]float64{
+			0:     45,
+			10800: 1045,
+		},
+		[]structs.QueryArithmetic{
+			{
+				LHS:        1,
+				ConstantOp: true,
+				Operation:  utils.LetLessThanOrEqualTo,
+				Constant:   1045,
+			},
+		})
+}
+
+func Test_GetResults_EQLC(t *testing.T) {
+	test_GetResults_Ops(t,
+		map[uint32]float64{
+			0:     45,
+			10800: 1045,
+		},
+		map[uint32]float64{
+			0: 45,
+		},
+		[]structs.QueryArithmetic{
+			{
+				LHS:        1,
+				ConstantOp: true,
+				Operation:  utils.LetEquals,
+				Constant:   45,
+			},
+		})
+}
+
+func Test_GetResults_NEQ(t *testing.T) {
+	test_GetResults_Ops(t,
+		map[uint32]float64{
+			0:     45,
+			10800: 1045,
+		},
+		map[uint32]float64{
+			10800: 1045,
+		},
+		[]structs.QueryArithmetic{
+			{
+				LHS:        1,
+				ConstantOp: true,
+				Operation:  utils.LetNotEquals,
+				Constant:   45,
+			},
+		})
+}
+
+func test_GetResults_Ops(t *testing.T, initialEntries map[uint32]float64, ansMap map[uint32]float64, queryOps []structs.QueryArithmetic) {
+	mQuery := &structs.MetricsQuery{
+		MetricName:  "test.metric.0",
+		HashedMName: 1,
+		Downsampler: structs.Downsampler{
+			Interval:   3,
+			Unit:       "h",
+			CFlag:      false,
+			Aggregator: structs.Aggreation{AggregatorFunction: utils.Avg},
+		},
+	}
+	qid := uint64(0)
+	metricsResults := mresults.InitMetricResults(mQuery, qid)
+	assert.NotNil(t, metricsResults)
+
+	var tsGroupId *bytebufferpool.ByteBuffer = bytebufferpool.Get()
+	defer bytebufferpool.Put(tsGroupId)
+	_, err := tsGroupId.Write([]byte("color:yellow"))
+	assert.NoError(t, err)
+	series := mresults.InitSeriesHolder(mQuery, tsGroupId)
+
+	// The entries' timestamp interval is larger than the data source interval (3 hours), so they can become separate key-value pairs.
+	for timestamp, val := range initialEntries {
+		series.AddEntry(timestamp, val)
+	}
```

Comment:
You don't know this comment is correct in general because `initialEntries` is a parameter and you haven't done any validation on it yet.

You might want to move the downsampler config out of this function (and pass it as a parameter) so that it's clear from the tests (e.g., `Test_GetResults_GTE()`) how downsampling is working. Or find a different way to organize this to make it more clear